manual/manual_circuit.rst

```diff
@@ -435,36 +435,168 @@ In addition :py:meth:`Circuit.get_unitary` can be used to numerically calculate
+
+We also define :math:`G`-depth (for a subset of gate types :math:`G`) as the minimum number of layers of gates in :math:`G` required to run the :py:class:`Circuit`, allowing for topological reorderings. Specific cases of this like :math:`T`-depth and :math:`CX`-depth are common to the literature on circuit simplification [Amy2014]_ [Meij2020]_.
+
+.. jupyter-execute::
+
+    from pytket import Circuit, OpType
+    from pytket.circuit.display import render_circuit_jupyter
+
+    circ = Circuit(3)
+    circ.T(0)
+    circ.CX(0, 1)
+    circ.CX(2, 0)
+    circ.add_gate(OpType.CnRy, [0.6], [0, 1, 2])
+    circ.T(2)
+    circ.CZ(0, 1)
+    circ.CZ(1, 2)
+    circ.T(1)
+
+    render_circuit_jupyter(circ) # draw circuit diagram
+
+    print("T gate count =", circ.n_gates_of_type(OpType.T))
+    print("#1qb gates =", circ.n_1qb_gates())
+    print("#2qb gates =", circ.n_2qb_gates())
+    print("#3qb gates =", circ.n_nqb_gates(3)) # count the single CnRy gate (n=3)
+    print("T gate depth =", circ.depth_by_type(OpType.T))
+    print("2qb gate depth =", circ.depth_by_type({OpType.CX, OpType.CZ}))
+
+.. note:: Each of these metrics will analyse the :py:class:`Circuit` "as is", so they will consider each Box as a single unit rather than breaking it down into basic gates, nor will they perform any non-trivial gate commutations (those that don't just follow by deformation of the DAG) or gate decompositions (e.g. recognising that a :math:`CZ` gate would contribute 1 to :math:`CX`-count in practice).
+
 Boxes
 -----
 
-.. Boxes abstract away complex structures as black-box units within larger circuits
-
 Working with individual basic gates is sufficient for implementing arbitrary circuits, but that doesn't mean it is the most convenient option. It is generally far easier to argue the correctness of a circuit's design when it is constructed using higher-level constructions. In ``pytket``, the concept of a "Box" is to abstract away such complex structures as black-boxes within larger circuits.
 
+Circuit Boxes and Controlled Unitaries
+======================================
+
+.. Boxes abstract away complex structures as black-box units within larger circuits
+
 .. Simplest case is the `CircBox`
 
 The simplest example of this is a :py:class:`CircBox`, which wraps up another :py:class:`Circuit` defined elsewhere into a single black-box. The difference between adding a :py:class:`CircBox` and just appending the :py:class:`Circuit` is that the :py:class:`CircBox` allows us to wrap up and abstract away the internal structure of the subcircuit we are adding so it appears as if it were a single gate when we view the main :py:class:`Circuit`.
 
+Lets first build a basic quantum circuit which implements a simplified version of a Grover oracle and then add
```

Comment:
```suggestion
Let's first build a basic quantum circuit which implements a simplified version of a Grover oracle and then add
```

Reply:
Done, apologies for the frequent grammatical errors it seems the spell checker on my IDE was only partially enabled.